firmware/wpilib-ws-romi/wpilib-ws-romi.ino

```diff
@@ -182,35 +261,67 @@ void configureBuiltins(uint8_t config) {
   rPiLink.buffer.builtinConfig = 0;
 }
 
-void configureIO(uint8_t config) {
-  uint8_t ioChannel = (config >> 2) & 0xF;
-  uint8_t mode = config & 0x3;
-
-  if (ioChannel < 0 || ioChannel > 4) return;
-  ioChannelModes[ioChannel] = mode;
+void configureIO(uint16_t config) {
+  // 16 bit config register
+  //
+  // MSB
+  // 0 | NEW CONFIG FLAG |
+  //   |-----------------|
+  // 1 |  Pin 0 Mode     |
+  // 2 |  ArdPin 11      |
+  //   |-----------------|
+  // 3 |  Pin 1 Mode     |
+  // 4 |  ArdPin 4       |
+  //   |-----------------|
+  // 5 |  Pin 2 Mode     |
+  // 6 |  ArdPin 20      |
+  //   |-----------------|
+  // 7 |  Pin 3 Mode     |
+  // 8 |  ArdPin 21      |
+  //   |-----------------|
+  // 9 |  Pin 4 Mode     |
+  // 10|  ArdPin 22      |
+  //   |-----------------|
+  // 11|  RESERVED       |
+  // 12|                 |
+  // 13|                 |
+  // 14|                 |
+  // 15|                 |
+  for (uint8_t ioChannel = 0; ioChannel < 5; ioChannel++) {
+    uint8_t offset = 13 - (2 * ioChannel);
+    uint8_t mode = (config >> offset) & 0x3;
+
+    // Disconnect PWMs
+    if (pwms[ioChannel].attached()) {
+      pwms[ioChannel].detach();
+    }
 
-  // Detach any attached servos
-  if (mode != kModePwm && pwms[ioChannel].attached()) {
-    pwms[ioChannel].detach();
-  }
+    ioChannelModes[ioChannel] = mode;
 
-  switch(mode) {
-    case kModeDigitalOut:
-      pinMode(ioDioPins[ioChannel], OUTPUT);
-      break;
-    case kModeDigitalIn:
-      pinMode(ioDioPins[ioChannel], INPUT);
-      break;
-    case kModePwm:
-      if (!pwms[ioChannel].attached()) {
+    switch(mode) {
+      case kModeDigitalOut:
+        pinMode(ioDioPins[ioChannel], OUTPUT);
+        break;
+      case kModeDigitalIn:
+        pinMode(ioDioPins[ioChannel], INPUT_PULLUP);
+        break;
+      case kModePwm:
         pwms[ioChannel].attach(ioDioPins[ioChannel]);
-      }
-      break;
+        break;
+      case kModeAnalogIn:
+        if (ioChannel > 0) {
+          // Make sure we set the pin back correctly
+          digitalWrite(ioAinPins[ioChannel], LOW);
```

Comment:
Why are we setting digitalWrite for analog in mode?

Reply:
Basically to make sure that the we're not in pullup mode (since writing HIGH on a digital input pin is the same as setting it to INPUT_PULLUP mode)

Reply:
I think that is goofy, but looks like that is the standard for the GPIO on the board.